src/shared/filter/all-exceptions.filter.spec.ts

```diff
@@ -1,12 +1,194 @@
+import { Test, TestingModule } from '@nestjs/testing';
 import { AllExceptionsFilter } from './all-exceptions.filter';
 import { ConfigService } from '@nestjs/config';
+import { HttpException, HttpStatus } from '@nestjs/common';
+import { REQUEST_ID_TOKEN_HEADER } from '../constant';
+import { v4 as uuidv4 } from 'uuid';
+import { AppLogger } from '../logger/logger.service';
+
+const mockMessage1 = 'mock exception string';
+const mockMessage2 = { hello: 'world', hi: 'joe' };
+const mockMessage3 = 'Somethingn is very wrong';
+
+const mockException1 = new HttpException(mockMessage1, HttpStatus.NOT_FOUND);
+const mockException2 = new HttpException(mockMessage2, HttpStatus.BAD_REQUEST);
+const mockError = new Error(mockMessage3);
 
 describe('AllExceptionsFilter', () => {
-  const mockedLogger = { setContext: jest.fn() };
+  let mockContext: any;
+  let mockRequest: any;
+  let mockResponse: any;
+
+  let mockConfigService = {
+    get: (key) => 'development',
+  };
+  let mockAppLogger = {
```

Comment:
Current tests call it `mockedLogger`.
In case we want to keep it consistent across all of our tests. Just a minor observation. :smile: